src/core/ModuleManager.sol

```diff
@@ -9,6 +9,32 @@ import { ResolverRecord, ModuleRecord, ResolverUID } from "../DataTypes.sol";
 import { ResolverManager } from "./ResolverManager.sol";
 import { IRegistry } from "../IRegistry.sol";
 
+/**
+ * In order to separate msg.sender context from registry,
+ * interactions with external Factories are done with this Trampoline contract.
+ */
+contract FactoryTrampoline {
```

Comment:
NOTE: added factory call trampoline, so calls made to factory dont come from msg.sender == registry